### DISCUSSION TITLE
Error at forecast from load model ForecastX and ForecastingGridSearchCV.

### DISCUSSION THREAD

**Discussion by akmalmvic0312** [alan-turing-institute/sktime#5328] [Q&A]
Hello,

I want to ask and discuss about my forecasting pipe using ForecastX. In ForecastX documents, i should to give fh at fitting parameter (gscv.fit(train, X=X_train, fh=fh). Then, i want to register the model to mlflow and load back to other python file that run the forecast one step a head from last data testing or data validation. But there is error in the (.predict(fh) :

_**Exception has occurred: ValueError
A different forecasting horizon fh has been provided from the one seen in fit. If you want to change the forecasting horizon, please re-fit the forecaster. This is because fitting of the ForecastingGridSearchCV depends on fh.**_ 

How to handle this error?

**Comment by fkiraly** [comment]
@yarnabrina, I vaguely remember you had the same or a similar issue a while ago - did we resolve that or did we end up reporting a bug?

**Reply by yarnabrina** [reply]
Are you thinking about this?

https://github.com/sktime/sktime/issues/4698

I don't think this was resolved.

I'm using ForecastX along with ForecastingRandomSearchCV, and didn't face any problem with save/load using dill/pickle (didn't try with mlflow).

But here the issue is with one step forecast part I think, there was a recent question in discord help-desk as well by Amank.

@akmalmvic0312 can you share a MCVE?

**Comment by fkiraly** [comment]
perhaps, do you have some code on dummy data to replicate the bug?
